integration/clean/clean_dedicated_host/clean_dedicated_host.go

```diff
@@ -0,0 +1,83 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+
+//go:build clean
+// +build clean
+
+package main
+
+import (
+	"context"
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/service/ec2"
+	"github.com/aws/aws-sdk-go-v2/service/ec2/types"
+	"log"
+	"time"
+)
+
+// Can't release a host if it was being used within the last 24 hr add 2 hr as a buffer
+const keepDurationDedicatedHost = -1 * time.Hour * 26
+const tagName = "tag:Name"
+const tagValue = "IntegrationTestMacDedicatedHost"
+
+func main() {
+	err := cleanDedicatedHost()
+	if err != nil {
+		log.Fatalf("errors cleaning %v", err)
+	}
+}
+
+func cleanDedicatedHost() error {
+	log.Print("Begin to clean EC2 Dedicated Host")
+
+	expirationDateDedicatedHost := time.Now().UTC().Add(keepDurationDedicatedHost)
+	cxt := context.Background()
+	defaultConfig, err := config.LoadDefaultConfig(cxt)
+	if err != nil {
+		return err
+	}
+	ec2client := ec2.NewFromConfig(defaultConfig)
+
+	dedicatedHosts, err := getDedicatedHost(cxt, ec2client)
+	if err != nil {
+		return err
+	}
+
+	dedicatedHostIds := make([]string, 0)
```

Comment:
Nit: I think it's weird to allocate a capacity of zero here. The maximum capacity we could have is `len(dedicatedHosts)`, so we could just always allocate an array of that size.

Reply:
If you do that you can't check if the list is empty. len(dedicatedHostIds) == 0  will always return false.

Reply:
I _thought_ that len and cap would return different values for `make([]string, 10)`, but I was wrong. Ignore this